subprojects/common/file-storage/src/main/kotlin/com/avito/filestorage/RemoteStorageFactory.kt

```diff
@@ -17,40 +13,19 @@ object RemoteStorageFactory {
 
     fun create(
         endpoint: String,
+        httpClientProvider: HttpClientProvider,
         loggerFactory: LoggerFactory,
         timeProvider: TimeProvider,
-        httpClient: OkHttpClient = getHttpClient(
-            logger = loggerFactory.create<RemoteStorage>(),
-            retryInterceptor = null,
-            verbose = false // do not enable for production, generates a ton of logs
-        )
+        testInterceptor: Interceptor? = null,
```

Comment:
This is an abstraction leak. You could wrap the `httpClientProvider` to return the client with applied testInterceptor

Reply:
I did that in first place, but it's also overcomplicated

Reply:
removed it